Sources/Host/Linux/PTrace.cpp

```diff
@@ -255,19 +255,16 @@ ErrorCode PTrace::suspend(ProcessThreadId const &ptid) {
 
 ErrorCode PTrace::step(ProcessThreadId const &ptid, ProcessInfo const &pinfo,
                        int signal, Address const &address) {
+#if defined(ARCH_ARM)
+  DS2BUG("single stepping for ARM must be implemented in software");
+#endif
```

Comment:
What if we make this a compilation error by commenting out the entire definition on ARM?